lms/djangoapps/grades/tests/test_signals.py

```diff
@@ -148,3 +158,74 @@ def test_score_reset_bad_user(self):
         self.get_user_mock = self.setup_patch('lms.djangoapps.grades.signals.user_by_anonymous_id', None)
         submissions_score_reset_handler(None, **SUBMISSION_RESET_KWARGS)
         self.signal_mock.assert_not_called()
+
+
+@ddt.ddt
+class ScoreChangedUpdatesSubsectionGradeTest(ModuleStoreTestCase):
+    """
+    Ensures that upon SCORE_CHANGED signals, the handler
+    initiates an update to the affected subsection grade.
+    """
+    def setUp(self):
+        super(ScoreChangedUpdatesSubsectionGradeTest, self).setUp()
+
+    def setUpCourseAndUser(self, enable_subsection_grades):
+        """
+        Configures the course for this test.
+        """
+        self.user = UserFactory()
+        self.course = CourseFactory.create(
+            org='edx',
+            name='course',
+            run='run',
+            metadata={'enable_subsection_grades_saved': enable_subsection_grades})
+        
+        self.chapter = ItemFactory.create(parent=self.course, category="chapter", display_name="Chapter")
+        self.sequential = ItemFactory.create(parent=self.chapter, category='sequential', display_name="Open Sequential")
+        self.problem = ItemFactory.create(parent=self.sequential, category='problem', display_name='problem')
+
+        self.score_changed_kwargs = {
+            'points_possible': 10,
+            'points_earned': 5,
+            'user_id': self.user.id,
+            'course_id': unicode(self.course.id),
+            'usage_id': unicode(self.problem.location),
+        }
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_subsection_grade_updated_on_signal(self, default_store):
+        with self.store.default_store(default_store):
+            self.setUpCourseAndUser(True)
+            with check_mongo_calls(2) and self.assertNumQueries(18):
```

Comment:
The number does not increase as more components are added. The exact queries are as follows:

`QUERY = u'SELECT "auth_user"."id", "auth_user"."password", "auth_user"."last_login", "auth_user"."is_superuser", "auth_user"."username", "auth_user"."first_name", "auth_user"."last_name", "auth_user"."email", "auth_user"."is_staff", "auth_user"."is_active", "auth_user"."date_joined" FROM "auth_user" WHERE "auth_user"."id" = %s' - PARAMS = (2,)

QUERY = u'SELECT "student_courseaccessrole"."id", "student_courseaccessrole"."user_id", "student_courseaccessrole"."org", "student_courseaccessrole"."course_id", "student_courseaccessrole"."role" FROM "student_courseaccessrole" WHERE "student_courseaccessrole"."user_id" = %s' - PARAMS = (2,)

QUERY = u'SELECT "courseware_studentmodule"."module_id", "courseware_studentmodule"."grade", "courseware_studentmodule"."max_grade" FROM "courseware_studentmodule" WHERE ("courseware_studentmodule"."course_id" = %s AND "courseware_studentmodule"."student_id" = %s AND "courseware_studentmodule"."module_id" IN (%s))' - PARAMS = (u'edx/course_0/run', 2, u'i4x://edx/course_0/problem/problem')

QUERY = u'SELECT "student_anonymoususerid"."id", "student_anonymoususerid"."user_id", "student_anonymoususerid"."anonymous_user_id", "student_anonymoususerid"."course_id" FROM "student_anonymoususerid" WHERE ("student_anonymoususerid"."course_id" = %s AND "student_anonymoususerid"."user_id" = %s)' - PARAMS = (u'edx/course_0/run', 2)

QUERY = u'SAVEPOINT "s139642533689088_x298"' - PARAMS = ()

QUERY = u'INSERT INTO "student_anonymoususerid" ("user_id", "anonymous_user_id", "course_id") VALUES (%s, %s, %s)' - PARAMS = (2, u'03852a25f4d45a00839b489d8868c7e5', u'edx/course_0/run')

QUERY = u'RELEASE SAVEPOINT "s139642533689088_x298"' - PARAMS = ()

QUERY = u'SELECT "submissions_scoresummary"."id", "submissions_scoresummary"."student_item_id", "submissions_scoresummary"."highest_id", "submissions_scoresummary"."latest_id", "submissions_studentitem"."id", "submissions_studentitem"."student_id", "submissions_studentitem"."course_id", "submissions_studentitem"."item_id", "submissions_studentitem"."item_type", "submissions_score"."id", "submissions_score"."student_item_id", "submissions_score"."submission_id", "submissions_score"."points_earned", "submissions_score"."points_possible", "submissions_score"."created_at", "submissions_score"."reset" FROM "submissions_scoresummary" INNER JOIN "submissions_studentitem" ON ( "submissions_scoresummary"."student_item_id" = "submissions_studentitem"."id" ) INNER JOIN "submissions_score" ON ( "submissions_scoresummary"."latest_id" = "submissions_score"."id" ) WHERE ("submissions_studentitem"."course_id" = %s AND "submissions_studentitem"."student_id" = %s)' - PARAMS = (u'edx/course_0/run', u'03852a25f4d45a00839b489d8868c7e5')

QUERY = u'SELECT "grades_persistentsubsectiongrade"."created", "grades_persistentsubsectiongrade"."modified", "grades_persistentsubsectiongrade"."id", "grades_persistentsubsectiongrade"."user_id", "grades_persistentsubsectiongrade"."course_id", "grades_persistentsubsectiongrade"."usage_key", "grades_persistentsubsectiongrade"."subtree_edited_timestamp", "grades_persistentsubsectiongrade"."course_version", "grades_persistentsubsectiongrade"."earned_all", "grades_persistentsubsectiongrade"."possible_all", "grades_persistentsubsectiongrade"."earned_graded", "grades_persistentsubsectiongrade"."possible_graded", "grades_persistentsubsectiongrade"."visible_blocks_hash" FROM "grades_persistentsubsectiongrade" WHERE ("grades_persistentsubsectiongrade"."course_id" = %s AND "grades_persistentsubsectiongrade"."user_id" = %s AND "grades_persistentsubsectiongrade"."usage_key" = %s)' - PARAMS = (u'edx/course_0/run', 2, u'i4x://edx/course_0/problem/problem')

QUERY = u'SAVEPOINT "s139642533689088_x299"' - PARAMS = ()

QUERY = u'SELECT "grades_visibleblocks"."id", "grades_visibleblocks"."blocks_json", "grades_visibleblocks"."hashed" FROM "grades_visibleblocks" WHERE "grades_visibleblocks"."hashed" = %s' - PARAMS = (u'5KRIQOKYvKycMaNUeoKTjQjaSKg=',)

QUERY = u'SAVEPOINT "s139642533689088_x300"' - PARAMS = ()

QUERY = u'INSERT INTO "grades_visibleblocks" ("blocks_json", "hashed") VALUES (%s, %s)' - PARAMS = (u'[{"locator":"i4x://edx/course_0/problem/problem","max_score":0,"weight":null}]', u'5KRIQOKYvKycMaNUeoKTjQjaSKg=')

QUERY = u'RELEASE SAVEPOINT "s139642533689088_x300"' - PARAMS = ()

QUERY = u'SELECT (1) AS "a" FROM "grades_visibleblocks" WHERE "grades_visibleblocks"."hashed" = %s LIMIT 1' - PARAMS = (u'5KRIQOKYvKycMaNUeoKTjQjaSKg=',)

QUERY = u'SELECT (1) AS "a" FROM "grades_persistentsubsectiongrade" WHERE ("grades_persistentsubsectiongrade"."course_id" = %s AND "grades_persistentsubsectiongrade"."usage_key" = %s AND "grades_persistentsubsectiongrade"."user_id" = %s) LIMIT 1' - PARAMS = (u'edx/course_0/run', u'i4x://edx/course_0/problem/problem', 2)

QUERY = u'INSERT INTO "grades_persistentsubsectiongrade" ("created", "modified", "user_id", "course_id", "usage_key", "subtree_edited_timestamp", "course_version", "earned_all", "possible_all", "earned_graded", "possible_graded", "visible_blocks_hash") VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)' - PARAMS = (u'2016-08-22 13:20:28.296284', u'2016-08-22 13:20:28.298234', 2, u'edx/course_0/run', u'i4x://edx/course_0/problem/problem', u'2016-08-22 13:20:27.282000', u'', 0.0, 0.0, 0.0, 0.0, u'5KRIQOKYvKycMaNUeoKTjQjaSKg=')

QUERY = u'RELEASE SAVEPOINT "s139642533689088_x299"' - PARAMS = ()`